src/main/java/com/denizenscript/denizencore/utilities/SimpleDefinitionProvider.java

```diff
@@ -6,7 +6,19 @@
 
 public class SimpleDefinitionProvider implements DefinitionProvider {
 
-    private final MapTag definitions = new MapTag();
+    private final MapTag definitions;
+
+    public SimpleDefinitionProvider(MapTag map) {
+        this.definitions = map;

```

Comment:
Thaaat's dangerous. This constructor should make a copy of the map, not track the original instance directly.